src/importer.cpp

```diff
@@ -0,0 +1,484 @@
+        for (const auto &entry : newComponentNames) {
+            std::string newName = entry.first;
+            size_t count = 1;
+            while (std::find(compNames.begin(), compNames.end(), newName) != compNames.end()) {
+                newName += "_" + convertToString(count++);
+            }
+            if (newName != entry.first) {
+                entry.second->setName(newName);
+            }
+        }
+
+        // If the component 'component' has variables then they are equivalent variables and they
+        // need to be exchanged with the real variables from the component 'importedComponent'.
+        for (size_t i = 0; i < component->variableCount(); ++i) {
+            auto placeholderVariable = component->variable(i);
+            for (size_t j = 0; j < placeholderVariable->equivalentVariableCount(); ++j) {
+                auto localModelVariable = placeholderVariable->equivalentVariable(j);
+                auto importedComponentVariable = importedComponentCopy->variable(placeholderVariable->name());
+                Variable::removeEquivalence(placeholderVariable, localModelVariable);
+                Variable::addEquivalence(importedComponentVariable, localModelVariable);
+            }
+        }
+        parent->replaceComponent(index, importedComponentCopy);
+
+        // Apply the rebased equivalence map onto the modified model.
+        applyEquivalenceMapToModel(rebasedMap, model);
+
+        // Copy over units used in imported component to this model.
+        std::map<std::string, std::string> unitsNamesToReplace;
+        for (const auto &u : requiredUnits) {
+            if (!model->hasUnits(u)) {
+                auto orignalName = u->name();
+                size_t count = 0;
+                while (!model->hasUnits(u) && model->hasUnits(u->name())) {
+                    auto name = u->name();
+                    name += "_" + convertToString(++count);
+                    u->setName(name);
+                }
+                model->addUnits(u);
+                if (orignalName != u->name()) {
+                    unitsNamesToReplace[orignalName] = u->name();
+                }
+            }
+        }
+        findAndReplaceComponentsCnUnitsNames(importedComponentCopy, unitsNamesToReplace);
+    }
+}
+
+void flattenComponentTree(const ComponentEntityPtr &parent, const ComponentPtr &component, size_t componentIndex)
+{
+    flattenComponent(parent, component, componentIndex);
+    auto flattenedComponent = parent->component(componentIndex);
+    for (size_t index = 0; index < flattenedComponent->componentCount(); ++index) {
+        auto c = flattenedComponent->component(index);
+        flattenComponentTree(flattenedComponent, c, index);
+    }
+}
+
+ModelPtr Importer::flatten(const ModelPtr &inModel)
+{
+    if (inModel->hasUnresolvedImports()) {
+        // Mimicking behaviour of previous flatten() function... not sure this is the way to go?
```

Comment:
Delete that comment. If it is not then we will need to create an issue to fix it.

Reply:
98f73054